entry/entry_io.go

```diff
@@ -1,86 +1,238 @@
 package entry // import "berty.tech/go-ipfs-log/entry"
 
 import (
-	"berty.tech/go-ipfs-log/iface"
-	"berty.tech/go-ipfs-log/io"
 	"context"
-	"fmt"
+	"sync"
+	"time"
+
 	"github.com/ipfs/go-cid"
+
+	"berty.tech/go-ipfs-log/iface"
+	"berty.tech/go-ipfs-log/io"
 )
 
 type FetchOptions = iface.FetchOptions
 
 // FetchParallel retrieves IPFS log entries.
 func FetchParallel(ctx context.Context, ipfs io.IpfsServices, hashes []cid.Cid, options *FetchOptions) []iface.IPFSLogEntry {
-	var entries []iface.IPFSLogEntry
+	entries := []iface.IPFSLogEntry(nil)
+
+	wg := sync.WaitGroup{}
+	wg.Add(len(hashes))
 
 	for _, h := range hashes {
-		entries = append(entries, FetchAll(ctx, ipfs, []cid.Cid{h}, options)...)
+		go func(h cid.Cid) {
+			defer wg.Done()
+
+			entries = append(entries, FetchAll(ctx, ipfs, []cid.Cid{h}, options)...)
```

Comment:
not always thread-safe, you may use an output channel and a dedicated goroutine to construct the unified slice

Reply:
changed for `fetchedEntries = make([][]iface.IPFSLogEntry, len(hashes))` instead of a chan which was easier to implement and read IMO